// @opentui/core/index-e4hzc2j2.js
// @bun
import {
  exports_src
} from "./index-nyw5p3ep.js";
import {
  __require
} from "./index-nkrr8a4c.js";

// src/runtime-plugin.ts
var CORE_RUNTIME_SPECIFIER = "@opentui/core";
var CORE_3D_RUNTIME_SPECIFIER = "@opentui/core/3d";
var CORE_TESTING_RUNTIME_SPECIFIER = "@opentui/core/testing";
var RUNTIME_MODULE_PREFIX = "opentui:runtime-module:";
var DEFAULT_CORE_RUNTIME_MODULE_SPECIFIERS = [
  CORE_RUNTIME_SPECIFIER,
  CORE_3D_RUNTIME_SPECIFIER,
  CORE_TESTING_RUNTIME_SPECIFIER
];
var DEFAULT_CORE_RUNTIME_MODULE_SPECIFIER_SET = new Set(DEFAULT_CORE_RUNTIME_MODULE_SPECIFIERS);
var isCoreRuntimeModuleSpecifier = (specifier) => {
  return DEFAULT_CORE_RUNTIME_MODULE_SPECIFIER_SET.has(specifier);
};
var loadCore3dRuntimeModule = async () => {
  return await import("./3d.js");
};
var loadCoreTestingRuntimeModule = async () => {
  return await import("./testing.js");
};
var escapeRegExp = (value) => {
  return value.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
};
var exactSpecifierFilter = (specifier) => {
  return new RegExp(`^${escapeRegExp(specifier)}$`);
};
var runtimeModuleIdForSpecifier = (specifier) => {
  return `${RUNTIME_MODULE_PREFIX}${encodeURIComponent(specifier)}`;
};
var resolveRuntimeModuleExports = async (moduleEntry) => {
  if (typeof moduleEntry === "function") {
    return await moduleEntry();
  }
  return moduleEntry;
};
var runtimeLoaderForPath = (path) => {
  if (path.endsWith(".tsx")) {
    return "tsx";
  }
  if (path.endsWith(".jsx")) {
    return "jsx";
  }
  if (path.endsWith(".ts") || path.endsWith(".mts") || path.endsWith(".cts")) {
    return "ts";
  }
  if (path.endsWith(".js") || path.endsWith(".mjs") || path.endsWith(".cjs")) {
    return "js";
  }
  return null;
};
var rewriteRuntimeSpecifiers = (code, runtimeModuleIdsBySpecifier) => {
  let transformedCode = code;
  for (const [specifier, moduleId] of runtimeModuleIdsBySpecifier.entries()) {
    const escapedSpecifier = escapeRegExp(specifier);
    transformedCode = transformedCode.replace(new RegExp(`(from\\s+["'])${escapedSpecifier}(["'])`, "g"), `$1${moduleId}$2`).replace(new RegExp(`(import\\s+["'])${escapedSpecifier}(["'])`, "g"), `$1${moduleId}$2`).replace(new RegExp(`(import\\s*\\(\\s*["'])${escapedSpecifier}(["']\\s*\\))`, "g"), `$1${moduleId}$2`).replace(new RegExp(`(require\\s*\\(\\s*["'])${escapedSpecifier}(["']\\s*\\))`, "g"), `$1${moduleId}$2`);
  }
  return transformedCode;
};
function createRuntimePlugin(input = {}) {
  const runtimeModules = new Map;
  runtimeModules.set(CORE_RUNTIME_SPECIFIER, input.core ?? exports_src);
  runtimeModules.set(CORE_3D_RUNTIME_SPECIFIER, loadCore3dRuntimeModule);
  runtimeModules.set(CORE_TESTING_RUNTIME_SPECIFIER, loadCoreTestingRuntimeModule);
  for (const [specifier, moduleEntry] of Object.entries(input.additional ?? {})) {
    runtimeModules.set(specifier, moduleEntry);
  }
  const runtimeModuleIdsBySpecifier = new Map;
  for (const specifier of runtimeModules.keys()) {
    runtimeModuleIdsBySpecifier.set(specifier, runtimeModuleIdForSpecifier(specifier));
  }
  return {
    name: "bun-plugin-opentui-runtime-modules",
    setup: (build) => {
      for (const [specifier, moduleEntry] of runtimeModules.entries()) {
        const moduleId = runtimeModuleIdsBySpecifier.get(specifier);
        if (!moduleId) {
          continue;
        }
        build.module(moduleId, async () => ({
          exports: await resolveRuntimeModuleExports(moduleEntry),
          loader: "object"
        }));
        build.onResolve({ filter: exactSpecifierFilter(specifier) }, () => ({ path: moduleId }));
      }
      build.onLoad({ filter: /\.(?:[cm]?js|[cm]?ts|jsx|tsx)$/ }, async (args) => {
        const loader = runtimeLoaderForPath(args.path);
        if (!loader) {
          return;
        }
        const file = Bun.file(args.path);
        const contents = await file.text();
        const transformedContents = rewriteRuntimeSpecifiers(contents, runtimeModuleIdsBySpecifier);
        return {
          contents: transformedContents,
          loader
        };
      });
    }
  };
}

export { isCoreRuntimeModuleSpecifier, runtimeModuleIdForSpecifier, createRuntimePlugin };

//# debugId=D871D36BF4DDA4A464756E2164756E21
